### ISSUE TITLE
Using explicit solution of equation for PD modeling?

### ISSUE THREAD

**Issue by karthikl6** [nlmixrdevelopment/nlmixr#145]
Hi @mattfidler ,

Can we use an explicit equation (Eg: Eff = baseline - slope*Cp) for PD modeling without ODE? If I want to do only PD modeling.

**Comment by mattfidler** [comment]
Yes you can have a PD only model

https://github.com/nlmixrdevelopment/nlmixr/issues/97

has an example, though it is only availble on github.

**Comment by karthikl6** [comment]
Thank you @mattfidler . But, when I try to add CWRES to the fit, R  shuts-off abruptly saying " R encountered fatal error". This happens only after this line  fit <- fit %>% addCwres().

`
myEmaxsaem <- function() {
   ini({

    temax <- log(0.8)
    ted50 <- log(10)
    tbaseline <- log(25)
    eta.emax ~ 0.09
    eta.ed50 ~ 0.09
    eta.baseline ~ 0.09
    add.err <- 1
  })
  model({
    
    emax <- exp(temax + eta.emax)
    ed50 <- exp(ted50 + eta.ed50)
    baseline <- exp(tbaseline + eta.baseline)
      
    eff =   (baseline * (1 - (((emax)* time)/((ed50)+time))))    
    eff ~ add(add.err)
  

  })
}

nlmixr(myEmaxsaem)

myEmaxsaemfit <- nlmixr(myEmaxsaem, mydat, est="saem",control=saemControl(print=500))

myEmaxsaemfit <- myEmaxsaemfit %>% addCwres()

``

**Comment by mattfidler** [comment]
Can you attach the data to this issue?  You can drag and drop the data into the issue. That way I can reproduce the issue.

**Comment by karthikl6** [comment]
I'm afraid I can't do that as the data is confidential. Is there any other way that I could get help on this??

**Comment by karthikl6** [comment]
I'm afraid I can't do that as the data is confidential. Is there any other way that I could get help on this??

**Comment by mattfidler** [comment]
Hi @karthikl6 

Since you provided the model, I think I fixed it, but I will be unsure until you verify.

Here is what I tested:

``` r

library(nlmixr)
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

set.seed(100)

m1  <- RxODE({
    temax <- log(0.8)
    ted50 <- log(10)
    tbaseline <- log(25)
    emax <- exp(temax + eta.emax)
    ed50 <- exp(ted50 + eta.ed50)
    baseline <- exp(tbaseline + eta.baseline)
    eff  <- (baseline * (1 - (emax* time)/(ed50+time)))+add.err
})

etaNames  <- c("eta.emax", "eta.ed50", "eta.baseline")
omega  <- diag(3)*0.1
dimnames(omega) <- list(etaNames,etaNames)

sigma  <- matrix(1,dimnames=list("add.err", "add.err"))

et  <- et(seq(0,48,length.out=24)) %>% et(id=1:50)

mydat  <- rxSolve(m1,et,sigma=sigma,omega=omega) %>%
    rename(dv=eff) %>% select(id,time,dv)

myEmaxsaem <- function() {
   ini({
    temax <- log(0.7)
    ted50 <- log(8)
    tbaseline <- log(15)
    eta.emax ~ 0.09
    eta.ed50 ~ 0.09
    eta.baseline ~ 0.09
    add.err <- 1
  })
  model({
    emax <- exp(temax + eta.emax)
    ed50 <- exp(ted50 + eta.ed50)
    baseline <- exp(tbaseline + eta.baseline)
    eff =   (baseline * (1 - (((emax)* time)/((ed50)+time))))
    eff ~ add(add.err)
  })
}

myEmaxsaemfit <- nlmixr(myEmaxsaem, mydat, est="saem",control=saemControl(print=500))
#> Compiling RxODE equations...
#> done.
#> 1:   -3.8766e-01   2.0638e+00   2.7006e+00   8.5500e-02   1.0924e-01   8.9904e-02   1.0000e+99
#> 500:   -3.8766e-01   2.0638e+00   2.7006e+00   8.0997e-02   1.0924e-01   8.9904e-02   1.0000e+99
#> Calculating residuals/tables
#> done.
#> Warning in .collectWarnings(do.call(foceiFit.data.frame0, call, envir =
#> parent.frame(1))): Some variances were NA/NaN, replaced with 1.
#> Warning in as.focei.saemFit(.fit, uif, pt, data = dat, calcResid =
#> calc.resid, : Covariance matrix non-positive definite, corrected by
#> sqrtm(fim %*% fim)

myEmaxsaemfit <- myEmaxsaemfit %>% addCwres()
#> Calculating residuals/tables
#> done.
#> Warning in .collectWarnings(do.call(foceiFit.data.frame0, call, envir =
#> parent.frame(1))): Some variances were NA/NaN, replaced with 1.
#> Warning in as.focei.saemFit(.saem, .uif, data = getData(fit), calcResid =
#> TRUE, : Covariance matrix non-positive definite, corrected by sqrtm(fim %*%
#> fim)
#> Can't find columns `WRES`, `CRES`, `CWRES`, `CPRED` in `.data`.

myEmaxsaemfit
#> ── nlmixr SAEM(PRED); OBJF not calculated fit ─────────────────────────────
#>  Gaussian/Laplacian Likelihoods: AIC(myEmaxsaemfit) or myEmaxsaemfit$objf etc.
#>  FOCEi CWRES & Likelihoods: addCwres(myEmaxsaemfit)
#> 
#> ── Time (sec; myEmaxsaemfit$time): ────────────────────────────────────────
#>   saem    setup optimize covariance table
#>  3.619 0.687793    3e-06      4e-06 0.037
#> 
#> ── Population Parameters (myEmaxsaemfit$parFixed or $parFixedDf): ─────────
#>             Est.     SE %RSE Back-transformed(95%CI) BSV(CV%) Shrink(SD)%
#> temax     -0.388 0.0802 20.7     0.679 (0.58, 0.794)    29.0%      35.8% 
#> ted50       2.06  0.049 2.37       7.88 (7.16, 8.67)    34.0%      39.9% 
#> tbaseline    2.7 0.0679 2.51           14.9 (13, 17)    30.7%      40.7% 
#> add.err      nan                                 nan                      
#> 
#>   Covariance Type (myEmaxsaemfit$covMethod): |fim|
#>   No correlations in between subject variability (BSV) matrix
#>   Full BSV covariance (myEmaxsaemfit$omega)
#>     or correlation (myEmaxsaemfit$omegaR; diagonals=SDs)
#>   Distribution stats (mean/skewness/kurtosis/p-value) available in $shrink
#> 
#> ── Fit Data (object myEmaxsaemfit is a modified tibble): ──────────────────
#> # A tibble: 1,200 x 16
#>   ID     TIME    DV  EVID  PRED   RES IPRED  IRES IWRES eta.emax eta.ed50
#>   <fct> <dbl> <dbl> <int> <dbl> <dbl> <dbl> <dbl> <dbl>    <dbl>    <dbl>
#> 1 1      0       NA     0  14.9    NA  16.9    NA    NA   -0.258   -0.261
#> 2 1      2.09    NA     0  12.8    NA  14.7    NA    NA   -0.258   -0.261
#> 3 1      4.17    NA     0  11.4    NA  13.3    NA    NA   -0.258   -0.261
#> # … with 1,197 more rows, and 5 more variables: eta.baseline <dbl>,
#> #   emax <dbl>, ed50 <dbl>, baseline <dbl>, eff <dbl>
```

<sup>Created on 2019-03-31 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by mattfidler** [comment]
You will have to update `RxODE` and `nlmixr`, of course:

```R
devtools::install_github("nlmixrdevelopment/RxODE")
devtools::install_github("nlmixrdevelopment/nlmixr")
```

**Comment by mattfidler** [comment]
Wait, I guess the object doesn't actually contain CWRES,  let me look into it a bit more.

**Comment by mattfidler** [comment]
It could be my simulated dataset;  You could check with your data-set still

**Comment by karthikl6** [comment]
Hi @mattfidler ,
This worked perfectly. Thank you very much!!

**Comment by karthikl6** [comment]
Can we still use mtime function that you've said before with this setup?? When I tried, it gave me this error:

RxODE doesn't know how to translate 'mtime' to SymPy.

**Comment by mattfidler** [comment]
Hi @karthikl6 its a incomplete solution.  I forgot to calculate the sensitivities for this parameter.

**Comment by karthikl6** [comment]
@mattfidler Thank you very much for letting me know about this. Do you have any estimated time-frame when this (mtime option) might be available?

**Comment by mattfidler** [comment]
I do not.

**Comment by mattfidler** [comment]
@karthikl6 the mtime feature requires either explicit jump sensitivites or numerical differencing;  This is a major change to nlmixr so the time-frame will probably be measured in months, or more.

**Comment by mattfidler** [comment]
This issue is still open for a explicit PD modeling;  I think it is really mtime for FOCEi.  For now I will close it and open another issue.
